tests/test_config.py

```diff
@@ -1,12 +1,80 @@
+from __future__ import annotations
 import os
 from pathlib import Path
+from typing import Any
 
 import pytest
 
 from starlette.config import Config, Environ, EnvironError
 from starlette.datastructures import URL, Secret
 
 
+def test_config_types() -> None:
```

Comment:
I added some tests for the type annotations. Open to feedback on how to make these nicer

Reply:
Maybe parametrize those:
```python
STR = [config("STR")]
is_str(STR)
```
And move the function definitions outside this test function (above) 🤔